### ISSUE TITLE
Replace the functionality of `ncap2` in the `quality.sh` script 

### ISSUE THREAD

**Issue by jasonfleming** [StormSurgeLive/asgs#1367]
The use of `ncap2` from NCO is prohibitive due to its dependencies. The functionality it provides will need to be replaced.

**Comment by wwlwpd** [comment]
Can you summarize what the following are doing - is this kind of like `awk` but for netCDF files?

From:

https://github.com/StormSurgeLive/asgs/blob/master/util/output/quality.sh

```
    # compute min, max, avg, stdev, and number of missing (i.e., dry) values
    # over the full domain for each time step and write to file
    for snap in $(seq 0 $((${filesNumDataSets[$file]} - 1))) ; do
        if [[ $file == "maxele.63.nc" || $file == "maxwvel.63.nc" || $file == "swan_HS_max.63.nc" ]]; then
            ncap2 -O -v -s "data[node]=0.0;data.set_miss(-99999.0);data=${netcdfVarName[$file]}(0:$(($np - 1)):1);print(data.min(),\"%f \");print(data.max(),\"%f \");print(data.avg(),\"%f \");print((data-data.avg()).rmssdn(),\"%f \");print(data.number_miss(),\"%d\n\");" $file statistics.nc >> statistics_${file}.txt
        else
            ncap2 -O -v -s "data[node]=0.0;data.set_miss(-99999.0);data=${netcdfVarName[$file]}($snap:$snap:1,0:$(($np - 1)):1);print(data.min(),\"%f \");print(data.max(),\"%f \");print(data.avg(),\"%f \");print((data-data.avg()).rmssdn(),\"%f \");print(data.number_miss(),\"%d\n\");" $file statistics.nc >> statistics_${file}.txt
        fi
    done
```

```
    "maxele.63.nc"|"fort.63.nc")
        # count the number of nodes that would be considered dry upon cold start based only on negative topobathy value
        numLand=$(ncap2 -O -v -s "land_msk[node]=0.0;land_msk.set_miss(-99999.0);where(depth < 0.0) land_msk=-99999.0; elsewhere land_msk=depth; print(land_msk.number_miss(),\"%d\n\");" $file statistics.nc)
```

**Comment by wwlwpd** [comment]
FWIW, antlr2 is a parser generator, so maybe it's possible to run the build command to some early intermediate step or maybe the code author can generate the intermediate files and not require anyone building it to do so? I'll email him.

**Comment by wwlwpd** [comment]
They want to be contacted through the forums, so here's my request - https://sourceforge.net/p/nco/discussion/9829/thread/2ab575e3ef/
